lib/ansible/modules/network/cloudengine/ce_stp.py

```diff
@@ -0,0 +1,969 @@
+#!/usr/bin/python
+#
+# This file is part of Ansible
+#
+# Ansible is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# Ansible is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+#
+
+ANSIBLE_METADATA = {'status': ['preview'],
+                    'supported_by': 'community',
+                    'version': '1.0'}
+
+DOCUMENTATION = '''
+---
+module: ce_stp
+version_added: "2.3"
+short_description: Manages STP configuration.
+description:
+    - Manages STP configurations on CloudEngine switches.
+author:
+    - wangdezhuang (@CloudEngine-Ansible)
+options:
+    state:
+        description:
+            - Specify desired state of the resource.
+        required: false
+        default: present
+        choices: ['present', 'absent']
+    stp_mode:
+        description:
+            - Set an operation mode for the current MSTP process.
+              The mode can be STP, RSTP, or MSTP.
+        required: false
+        default: null
+        choices: ['stp', 'rstp', 'mstp']
+    stp_enable:
+        description:
+            - Enable or disable STP on a switch.
+        required: false
+        default: null
+        choices: ['enable', 'disable']
+    stp_converge:
+        description:
+            - STP convergence mode.
+              Fast means set STP aging mode to Fast.
+              Normal means set STP aging mode to Normal.
+        required: false
+        default: null
+        choices: ['fast', 'normal']
+    bpdu_protection:
+        description:
+            - Configure BPDU protection on an edge port.
+              This function prevents network flapping caused by attack packets.
+        required: false
+        default: null
+        choices: ['enable', 'disable']
+    tc_protection:
+        description:
+            - Configure the TC BPDU protection function for an MSTP process.
+        required: false
+        default: null
+        choices: ['enable', 'disable']
+    tc_protection_interval:
+        description:
+            - Set the time the MSTP device takes to handle the maximum number of TC BPDUs
+              and immediately refresh forwarding entries.
+              The value is an integer ranging from 1 to 600, in seconds.
+        required: false
+        default: null
+    tc_protection_threshold:
+        description:
+            - Set the maximum number of TC BPDUs that the MSTP can handle.
+              The value is an integer ranging from 1 to 255. The default value is 1.
```

Comment:
I don't see `default=1` in the argspec, or in the documentation. Is this the default on the switch?

If so then this comment may need updating